### ISSUE TITLE
I deleted the base homeassistant container, how can I reinstall it?

### ISSUE THREAD

**Issue by carmolim** [home-assistant/hassio-installer#16]
I was following other tutorial, but it went wrong in the end, so I removed everything, and now I can't make it work again, need some help to reinstall the homeassistant container. 

Thanks.

**Comment by ifol** [comment]
I'm planning to reinstall by removing everything (all docker containers) and installing again too because of changing armhf containers to armv7. Did you remove the supervisor also and it is failing to reinstall homeassistant from scratch? I'd hate to run into the same issue and end up with non-working Hass.io

**Comment by unlucio** [comment]
@carmolim short answer: you can't (and welcome to my nightmare)
I've been in that situation a couple of times, the "supervisor" kind of fails a tad it's "supervising" and "orchestrating" duty when it comes to the `homeassistant` container.
From what I saw it will manage the add-ons but if `homeassistant` is deleted the only real solution is (regrettably) `rm -fr var/lib/docker/*`
I did add  a footnote at the end of the readme (https://github.com/home-assistant/hassio-installer#warning-do-not-delete-created-containers) hoping to save others from my same situation but this thread I guess is a testimony to my failure 😅

**Comment by ifol** [comment]
@unlucio I've read the readme note but there are situations when you want/need to just get rid of everything and start from scratch, I'd never imagine it would be such a hassle. The note in the readme doesn't help when you do want to remove Hass.io

But if I understand you correctly then I should be able to do:

docker containers prune
rm -rf /var/lib/docker/*
run the installer script

and I should be successful in reinstalling Hass.io?

**Comment by unlucio** [comment]
@ifol yes I totally agree there are situations when you need to zap all the containers, and I've been there as well, and yes I was quite baffled as well staring at my `docker ps` for ages waiting for a `homeassistant` container than never showed up.

Well, the main point of my note was to make people aware of the problem and perhaps give a possible safe line to avoid id (ie: the filter config), but I do think the thing needs to be actually better addressed in the supervisor (but I'm not an homeassistant developer).

And yup: that's pretty much it.
Just remember to stop the docker daemon before you zap the lib dir. I wouldn't be surprised if the daemon wouldn't appreciate having the rug pulled from under its feet. Here are the docker steps:
```sh
$ service docker stop
$ rm -fr /var/lib/docker/*
$ service docker start
```

**Comment by unlucio** [comment]
On a side note: I think all this will never be a priority for hass/homeassistant as they're clearly "betting' on hass-os more than custom installations.
 I can understand why they'd do that: they're likely to need less support hassle with a "locked" unified system VS having to deal with all the little possible quirks that can come out of custom installs.

That said: Imho a process that doesn't take care of creating the needed dependent resources cannot actually be called "supervisor" nor "orchestrator". Call it "master process" but "supervisor" in this case is misleading and imho the source of this confusion.

I was actually toying around with the idea of deploying hass.io in a kubernetes environment, but with things like this, it's pretty impossible :(

**Comment by unlucio** [comment]
and there we go: I just had to zap all my containers again :(

I had to restart my pi, and for some reason, homeassistant was presenting the welcome screen and all my configs were gone.
Now tailing the logs on that container is clearly an exercise in futility as it's going to play back weeks of logs.
The only reasonable path would be removing the homeassistant container in order to tail the logs form a fresh one and find the error, but it's clearly not possible for the discussed limitations.

**Comment by ciB89** [comment]
I tried this but the homeassistant docker container does not show up..??

**Comment by larstobi** [comment]
This just happened to me as I was unfortunate enough to click on the update button in Hassio. What happened was that the "supervisor" stopped and removed the homeassistant container, but never created and ran a new one. So, don't click on the update button in ther web ui!

**Comment by carmolim** [comment]
@unlucio, I tried to follow those steps, but didnt work. 

Than I followed these steps: [https://askubuntu.com/questions/935569/how-to-completely-uninstall-docker](url), reninstalled docker-ce and now is kind of working

The welcome screen was asking my user and password, the onboarding process didnt happen. And I was unable to install the addons, this error message appered:

``` 
Can't create container from addon_a0d7b954_nodered: 404 Client Error: Not Found 
```

Will keep trying.
